optimize/upgrade/src/main/java/io/camunda/optimize/upgrade/main/UpgradeMain.java

```diff
@@ -82,7 +83,17 @@ public static void main(final String... args) {
       for (final UpgradePlan upgradePlan : upgradePlans) {
         upgradeProcedure.performUpgrade(upgradePlan);
       }
-      upgradeProcedure.schemaUpgradeClient.initializeSchema();
+
+      final boolean initSchemaEnabled =
+          ConfigurationServiceBuilder.createDefaultConfiguration()
+              .getElasticSearchConfiguration()
+              .getConnection()
+              .isInitSchemaEnabled();
+      if (initSchemaEnabled) {
+        upgradeProcedure.schemaUpgradeClient.initializeSchema();
+      } else {
+        LOG.info("Schema initialization disabled, skipping");
+      }
```

Comment:
So I thought about this a little more and actually think we should approach upgrades a little bit differently. My thinking is basically that without these permissions, we cannot support upgrades. We rely heavily on async operations during the upgrade and do not want to proceed without the previous "step" being completed. I know that for this customer we are already descoping Upgrade support, so this is acceptable.

With this in mind, I would like us to handle this with an exception rather than a log. Specifically, before we actually perform the upgrade, we should ensure that the two new config options you added in this PR are enabled. If any of them are not enabled, I think we should fail early and throw an UpgradeException.

I also think that this simplifies things here in a way

Reply:
good point, implemented